tests/base/src/test/java/org/keycloak/test/admin/SMTPConnectionVaultTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.test.admin;
+
+import jakarta.ws.rs.core.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.models.AdminRoles;
+import org.keycloak.models.Constants;
+import org.keycloak.test.framework.annotations.InjectKeycloakUrls;
+import org.keycloak.test.framework.annotations.InjectRealm;
+import org.keycloak.test.framework.annotations.KeycloakIntegrationTest;
+import org.keycloak.test.framework.mail.MailServer;
+import org.keycloak.test.framework.mail.annotations.InjectMailServer;
+import org.keycloak.test.framework.realm.ManagedRealm;
+import org.keycloak.test.framework.realm.RealmConfig;
+import org.keycloak.test.framework.realm.RealmConfigBuilder;
+import org.keycloak.test.framework.server.KeycloakServerConfig;
+import org.keycloak.test.framework.server.KeycloakServerConfigBuilder;
+import org.keycloak.test.framework.server.KeycloakUrls;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@KeycloakIntegrationTest(config = SMTPConnectionVaultTest.SMTPVaultConfig.class)
+public class SMTPConnectionVaultTest {
+
+    @InjectRealm(config = SMTPRealmWithClientAndUser.class)
+    private ManagedRealm managedRealm;
+
+    private RealmResource realm;
+
+    @InjectMailServer
+    private MailServer mailServer;
+
+    @InjectKeycloakUrls
+    private KeycloakUrls keycloakUrls;
+
+    private Keycloak keycloak;
+
+    public static class SMTPVaultConfig implements KeycloakServerConfig {
+
+        @Override
+        public KeycloakServerConfigBuilder configure(KeycloakServerConfigBuilder config) {
+            URL url = Thread.currentThread().getContextClassLoader().getResource("vault");
```

Comment:
You should be able to just do something like this which is a bit cleaner:

```suggestion
            URL url = SMTPConnectionVaultTest.class.getResource("/vault");
```

Reply:
Even better if you put the files in `resources/java/org/keycloak/test/admin` then you can just do getResource("vault"), and the files are more associated with the actual tests using it than in the root dir.